src/App.tsx

```diff
@@ -0,0 +1,221 @@
+import { DetailsPanel } from "./weatherDetails/DetailsPanel";
+import { CurrentWeatherPanel } from "./currentWeather/CurrentWeatherPanel";
+import { createRoot } from "react-dom/client";
+import { useEffect, useState } from "react";
+import { DotSpinner } from "@uiball/loaders";
+
+const baseUrl = "http://api.openweathermap.org";
+const apiKey = "52cd117b0f18fb74a6f94c5c52c15753";
+
+type GcsValues = {
+  lat: string;
+  lon: string;
+};
+
+export type CurrentWeather = {
```

Comment:
You basically creates deadlock dependencies in your code and spaghetti code, because child components of app import types from it for some reason

![CleanShot 2023-11-24 at 17 16 00@2x](https://github.com/Jijuzo/Weather-App/assets/55194394/7fdc7bda-515d-41bf-a1fc-0264318485c3)